lib/ex_unit/lib/ex_unit/case.ex

```diff
@@ -552,8 +559,13 @@ defmodule ExUnit.Case do
 
     opts = Module.get_attribute(module, :ex_unit_module, [])
     async? = Keyword.get(opts, :async, false)
+    async_partition_key = Keyword.get(opts, :async_partition_key, nil)
     parameterize = Keyword.get(opts, :parameterize, nil)
 
+    if async_partition_key != nil and async? == false do
+      raise ArgumentError, ":async_partition_key is only a valid option when async: true"
+    end
```

Comment:
I am not sure we want this validation, because sometimes I may make the test sync only to validate something, and now I have to remove the group/partition-key option too. What if we simply call this `:group` and we say that tests in the same group are always `sync`?

Reply:
That's a nice solution, will do it like that then. 👍

Reply:
Removed the validation in https://github.com/elixir-lang/elixir/pull/13897/commits/8707d42c2794a58b6ab50da13ad6129217c1e584.